src/Pos/Util/UserSecret.hs

```diff
@@ -134,7 +184,19 @@ writeUserSecretRelease u
 
 -- | Helper for writing secret to file
 writeRaw :: UserSecret -> IO ()
-writeRaw u = BSL.writeFile (u ^. usPath) $ encode u
+writeRaw u = do
+    let path = u ^. usPath
+    -- On POSIX platforms, openTempFile guarantees that the file
+    -- will be created with mode 600.
```

Comment:
you probably mean openBinaryTempFile, and openTempFile is actually using 0o666 as default mask + whatever the process umask is.